tests/PHPStan/Analyser/data/constant-string-unions.php

```diff
@@ -33,6 +33,28 @@ public function unionOnRight(string $name, ?int $gesperrt = null, ?int $adaid =
 		assertType("'branch-a general'|'branch-b branch-a general'|'branch-b general'|'general'", $string);
 	}
 
+	public function unionOnBoth():void
+	{
+		$left = rand() ? 'a' : 'b';
+		$right = rand() ? 'x' : 'y';
+		assertType("'ax'|'ay'|'bx'|'by'", $left . $right);
+	}
+
+	public function encapsedString():void
+	{
+		$str = rand() ? 'a' : 'b';
+		$int = rand() ? 1 : 0;
+		$float = rand() ? 1.0 : 2.0;
+		$bool = (bool) rand();
+		$nullable = rand() ? 'a' : null;
+		assertType("'.a.'|'.b.'", ".$str.");
+		assertType("'.0.'|'.1.'", ".$int.");
+		assertType("'.1.'|'.2.'", ".$float.");
+		assertType("'..'|'.1.'", ".$bool.");
+		assertType("'..'|'.a.'", ".$nullable.");
+		assertType("'.a.0.'|'.a.1.'|'.b.0.'|'.b.1.'", ".$str.$int.");
```

Comment:
Can you please also provide an example where the logic goes over the limit and the type is simplified again? Thank you.